catalog/services_state_test.go

```diff
@@ -205,30 +205,31 @@ func Test_ServicesStateWithData(t *testing.T) {
 			})
 
 			Convey("Doesn't retransmit when the state is the same", func() {
-				state.Broadcasts = make(chan [][]byte, 1)
+				state.Broadcasts = make(chan [][]byte, 2)
 				state.AddServiceEntry(svc)
+
+				<-state.Broadcasts
+
 				svc.Updated = svc.Updated.Add(1 * time.Second)
 				state.AddServiceEntry(svc)
 
-				pendingBroadcast := false
-				select {
-				case <-state.Broadcasts:
-					pendingBroadcast = true
-				default:
-				}
-				So(pendingBroadcast, ShouldBeFalse)
+				time.Sleep(5 * time.Millisecond)
+
+				So(len(state.Broadcasts), ShouldEqual, 0)
```

Comment:
Looking at this more closely, let's delete this test. This is not desirable behavior.